command/list.go

```diff
@@ -118,5 +131,8 @@ func (c *ListCommand) Run(args []string) int {
 		return 2
 	}
 
+	// Hijack the secret's data to show the list with information.
+	secret.Data[doListWithInfoConstant] = c.flagDetailed
```

Comment:
I am not a huge fan of modifying the response data to add a flag meant to signal how `OutputList` should work. It might be cleaner to branch on `c.flagDetailed` here rather and potentially add a separate function like `OutputListWithDetails` rather than pushing the decision further down. It adds a bit of cognitive overhead as there is now a somewhat hidden flag.

Reply:
@ccapurso Some thoughts:

1. What do you think about avoiding the `-detailed` flag altogether and just showing the extra info always, if the API endpoint returns it? The one argument in favor (of `-detailed`) might be the JSON format though... If we do that, we wouldn't need the changes here, and it'd make some of the detection logic simpler. The other reason in favor of that is that we wouldn't have to fork _all_ of the List functions (you'd have to fork `OutputList` -> `OutputListWithDetails`, then all of the formatters would have to again be updated to have two methods, mostly duplicating the code except for the extra code we add for Outputs.
2. The other option might be if `c.flagDetailed == false`, we could just do a `delete(secret.Data, "key_info")` and that'd be simpler. Then the details can be unconditionally and we wouldn't need to add the extra flag. Other callers would benefit from it the same way.

The other thought was I was thinking was that it might be neat if we pushed this down server side (i.e., set that flag server side for the `/issuers` and `/keys` endpoints), but that might be an argument for removing the `-detailed` flag altogether and having it just be on the `ListResponseWithInfo`.

Perhaps @ivana-mcconnell has some thoughts on the matter, too?

Reply:
I imagine that the flag is necessary at this point given that changing the format of the output will introduce a backward incompatible change. As you mention, this is important for the JSON format. If someone pipes to `jq` currently when using `-format=json` for one of these endpoints they will instead get a map whereas before they got an array. In other words, I think an output change like this needs to be opt-in.

I agree, having to fork all the formatters would not be ideal. I wonder if it might make sense to add a flag to the `VaultUI` struct that basically acts as the `vault_cli_do_list_with_info` field. What do you think?

Reply:
@ccapurso That makes sense. I realize I have no idea how the `ui.format` field gets set, so I had to do it manually in `list.go`. If you know how that gets done, let me know and I can update it.

Reply:
One reason that I was thinking about having it on the `ui` level is that _maybe_ we would want to support a more expanded output for other commands in the future, similar to the `-o wide` option in `kubectl`. Not sure if others agree, just an idea, of course!
 
In terms of where `ui.format` is set, see [setupEnv](https://github.com/hashicorp/vault/blob/62415ecc4c880b98f9e170efc425c76d3ba3de24/command/main.go#L28).

Reply:
Oh huh, TIL. Does that mean https://github.com/hashicorp/vault/blob/62415ecc4c880b98f9e170efc425c76d3ba3de24/command/base.go#L57 and https://github.com/hashicorp/vault/blob/62415ecc4c880b98f9e170efc425c76d3ba3de24/command/base.go#L515-L525 are unused?

Reply:
That is a really good question. I think you can actually use `c.flagFormat` or `Format(c.UI)` to determine the output format. I think that I've used both. I think theoretically they should be same. It looks like `Format(c.UI)` is far more widely used.

Reply:
Okie dokie, I'll merge things together then :-)